### DISCUSSION TITLE
Notebooks for all May meeting 5/16  @ 9am PST

### DISCUSSION THREAD

**Discussion by tonyfast** [Iota-School/notebooks-for-all#109] [Announcements]
* Calendar link: [May 16, 2024 9-10:30 AM PST ](https://calendar.google.com/calendar/event?action=TEMPLATE&tmeid=NGlucTNiaHZhaWEwNThyZnA5N3Zmb2dqdmVfMjAyNDA1MTRUMTcwMDAwWiAzMzRjMjQxMTI1ZDU5NGFlMzIyMjI5N2NkZDkxMDZmZGE1NTUxMTY2MDBkMTg3MzMyODRiY2E5YmZlZWI4Zjg1QGc&tmsrc=334c241125d594ae3222297cdd9106fda555116600d18733284bca9bfeeb8f85%40group.calendar.google.com&scp=ALL)
* [zoom link](https://zoom.us/j/91857677315?pwd=Yzh5b1U5MUsraUhVM3NuajNoSVRldz09)

 <details>
      <summary>full zoom meeting information</summary>
       
Join Zoom Meeting
[https://zoom.us/j/91857677315?pwd=Yzh5b1U5MUsraUhVM3NuajNoSVRldz09](https://www.google.com/url?q=https://zoom.us/j/91857677315?pwd%3DYzh5b1U5MUsraUhVM3NuajNoSVRldz09&sa=D&source=calendar&usd=2&usg=AOvVaw0uwfUUJpRoKaR1ndtyWMQh)

Meeting ID: 918 5767 7315
Passcode: 792407

---

One tap mobile
+12532158782,,91857677315# US (Tacoma)
+12532050468,,91857677315# US

---

Dial by your location
• +1 253 215 8782 US (Tacoma)
• +1 253 205 0468 US
• +1 669 444 9171 US
• +1 669 900 9128 US (San Jose)
• +1 719 359 4580 US
• +1 346 248 7799 US (Houston)
• +1 312 626 6799 US (Chicago)
• +1 360 209 5623 US
• +1 386 347 5053 US
• +1 507 473 4847 US
• +1 564 217 2000 US
• +1 646 558 8656 US (New York)
• +1 646 931 3860 US
• +1 689 278 1000 US
• +1 301 715 8592 US (Washington DC)
• +1 305 224 1968 US
• +1 309 205 3325 US

Meeting ID: 918 5767 7315

Find your local number: [https://zoom.us/u/ac6hrypyWt](https://www.google.com/url?q=https://zoom.us/u/ac6hrypyWt&sa=D&source=calendar&usd=2&usg=AOvVaw1zPJmJV-H6lXMk15xNInuW)

</details>

## agenda

* [9:05 - 9:10] Welcoming and meeting goals
* [9:10 - 9:20] Voluntary introductions and short shares
  * introduce yourself and/or share your accessibility wins or something you found interesting this week
* [9:20 - 9:40] The disabled developer or data science experience

  we'll have a group discussion about the experiences of affected users of assistive technology. we want to hear about the daily experiences of disabled scientists and technologists to learn what accomodations are effective.

* [9:40 - 9:55] How do we drive better assistive technology experiences in open source

  it is important for disabled voices to be heard in designing software. in this discussion we'll work to understand what support disabled developers or wish existed so they could participate in open source software practices.

* [9:55 - 10:00] Wrap up 
  * What access needs should be consider in the future?
  * Suggestions for next month
* [10:00 - 10:30] Unstructured community time

**Comment by maddieamie** [comment]
# Minutes Notes - Informal

## Discussion of Navigation in Google Co-Lab, VS Code, Jupyter Notebook

- When you open a page, does it tell you how many cells there are?
- To Navigate: 
  - Using the 'e' keyboard shortcut to jump between cells to each editable area, and down arrow jumps out of editable area
- It should play sound when the code executes, or when things move around.
- There is an extension of VS Code that has two sounds when the code executes: one for successful execution and one for an error
- Is there a sound for when the code crashes? Does that use the error sound?
- In Tony's Jupyter Notebook, each cell is a form element, so it should have similar navigational functionality. 
- shortcuts discussed: 'e' as moving to editable areas, 'h' for headings

## How do people generally navigate in the browser?
- When first opening a new webpage, two people agreed they both do this coincidentally: 
  - First thing in browse mode, look at heading number 1. 
  - Use 'h' shortcut to navigate through all the headings and get an overview of how many there are
  - then use the arrow keys to navigate
- In Gmail, you can use the 'h' to go from email to email
  - Maybe for Jupyter Notebook, it would be nice to use headings to navigate between notebooks in browse mode, as if you were using different tabs in a browser?

## Navigating a Form
- Will often use 'e' shortcut to navigate to each part of the form
- 'a' to move through radio boxes
- Can use 'tab' to navigate and get a mental layout, then 'e' to go back to the editable part of the form
- A small discourse on browse mode vs. focus mode
  - browse mode: looking at all the things, getting a mental landscape on how to proceed
  - focus mode: generally used when making content
  - Often, folks will navigate through once and then edit the page

## Discussion of summarizing content on Tony's Jupyter Notebooks 
- Tony has a Notebook summary button, you can use 'tab' six times from the top of the page to navigate to it
- There are aria summaries that are used sometimes
- It could be nice to have a shortcut option when navigating to have a summary of content option
  - Maybe: 1st 'tab' is "skip to main content" button, 2nd 'tab' could be an option to jump to the summary button
  - K uses 'Control - F', then types 'sum' for 'summary' to quickly find the summary content
  - There is a screen-reader shortcut 'c' that jumps between tabs

## Discussion of keyboard shortcut differentiations 
- NVDA and JAWS both have different shortcuts to jump between tabs
- Often, E will look at the NVDA user manual to see new updates 
- T mentions that it seems like there are a lot of screen-reader updates that people have to adjust to
- E agrees that there are a lot of updates, now that it is mentioned
- Google recently stopped using HTML markers, and now they use their own shortcuts that don't align with screen-reader navigation shortcuts on the keyboard
- Wish companies would stick to a template that is more standardized, or aligned with screen-reader standard
  - It seems like so many apps have their own keyboard shortcuts that interfere with general navigation shortcuts and things used with accessibility software
  - JAWS feels better to one person when using Microsoft products often
  - Dolphin or Supernova in the UK is a common one when using Windows systems
    - It has a magnification focus, similar to a zoom

## ChatGPT Accessibility 
- E uses ChatGPT to ask what software uses which keyboard shortcut, since it is often hard to navigate to the help menu in most software to figure that out (they pay for 4.0, so maybe could be different for 3.5 in accuracy)
- You can also ask it how to install certain libraries
- In general, it is difficult to navigate the interface for ChatGPT
  - It doesn't have a break between the question and answer, it is all just one big text box
  - The bots of different versions of ChatGPT are unlabeled buttons, so they are not navigable 
  - It would be nice to be able to use a keyboard shortcut to jump back to the top of the question, since the focus moves down the screen as it generates text
  - If you navigate to a different page, it will lose the focus of where you were in the question. When you navigate back, it will put you back to the edit box on the page and you lose where you were in the answer that you were referencing. 
- Pro tip: When asking ChatGPT to generate an answer, E asks it to use headings, so you can tab through the answer with 'h'
- T says you can ask it to generate an answer in Markdown as well
- E mentions that it can generate in math languages too
- M never thought to specify the output type when asking it a question, mind-blown
- Such a powerful tool, but the interface is far from accessible 

## Data Accessibility
- What kinds of useful alt text are there?
  - almost too general of a question, it really depends on the context
  - AI generated alt text is not too terrible
- In regards to data, E would always want access to the raw data available as an option, even if they don't use it 
- K recommended Olly library through MIT: [Olli](https://mitvis.github.io/olli/)
- ChatGPT describes data pretty well
- Gemini - you can paste in an image and have the AI describe it to you
- JAWS/NVDA have an AI add-on to describe content, but they may not be free

**Comment by tonyfast** [comment]
i've posted a video from the event with chapters to the relevant sessions. one of my favorite quotes is towards the end by khalid saying:

>   "i don't think alt text is sufficient for someone who wants to do data science"

attendees: khalid k, @ecmandell, @maddieamie, @tonyfast 

https://www.youtube.com/watch?v=3pag8E8EuMg&feature=youtu.be
